### PR TITLE
Added storage of client id

### PR DESCRIPTION
**Pull request by gabrielonilsson**

### CODE REVIEW

File: ClientIdentifier.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Identifier } from "authly"
+
+export namespace ClientIdentifier {
+
+	export function getClientId() {
```
Comment (simonmika) [inline ClientIdentifier.ts]: There is something of with the naming of things here. Maybe use a static class with at static property?

File: ClientIdentifier.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+		let clientId = localStorage.getItem("clientId")
```
Comment (simonmika) [inline ClientIdentifier.ts]: Nothing is ever stored in localStorage.

**Review by simonmika — CHANGES_REQUESTED** [review]

File: ClientIdentifier.ts (hunk 2 above)
Comment (gabrielonilsson) [reply ClientIdentifier.ts]: I thought that was the plan. How should it be stored?

Comment (simonmika) [reply ClientIdentifier.ts]: If you generate a new one make sure to store it.

File: ClientIdentifier.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { Identifier } from "authly"
+
+export abstract class ClientIdentifier {
+	private constructor() {}
+	static set id(id: string) {
+		ClientIdentifier.id = id
```
Comment (simonmika) [inline ClientIdentifier.ts]: This does not work it is a infinite recursive call.

Comment (simonmika) [inline ClientIdentifier.ts]: Set should trigger local storage. If set should be public at all.

File: ClientIdentifier.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { Identifier } from "authly"
+
+export abstract class ClientIdentifier {
+	private constructor() {}
+	static set id(id: string) {
```
Comment (simonmika) [inline ClientIdentifier.ts]: Call the property `value` instead or call the class `Client` and the property `identifier`.

Comment (simonmika) [reply ClientIdentifier.ts]: We need to decide if we call it `id` or `identifier`. It should be one or the other, never both. @fredrikbryntesson

**Review by simonmika — CHANGES_REQUESTED** [review]

File: ClientIdentifier.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { Identifier } from "authly"
+
+export abstract class ClientIdentifier {
+	private constructor() {}
+	private static value: string
+	get value(): string {
```
Comment (simonmika) [inline ClientIdentifier.ts]: Should be `static`.

File: ClientIdentifier.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { Identifier } from "authly"
+
+export abstract class ClientIdentifier {
+	private constructor() {}
+	private static value: string
```
Comment (simonmika) [inline ClientIdentifier.ts]: Needs different name in order not to clash. What about `valueCache`?

File: ClientIdentifier.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+	get value(): string {
+		ClientIdentifier.value = localStorage.getItem("clientIdentifier")
+		if (!ClientIdentifier.value) {
+			ClientIdentifier.value = Identifier.generate(10)
```
Comment (simonmika) [inline ClientIdentifier.ts]: A 60-bit random number seams to overdo it but why not.

**Review by simonmika — CHANGES_REQUESTED** [review]

**Review by simonmika — APPROVED** [review]

**Pull request merged**